### ISSUE TITLE
Sending SMS via SNS without topic fails

### ISSUE THREAD

**Issue by gertjanmaas** [spulec/moto#2198]
Moto version: 1.3.8
Boto3 version: 1.9.42

Example:
```
import boto3
from moto import mock_sns
mock = mock_sns()
mock.start()
client = boto3.client('sns')
phone_number = "<insert valid E164 formatted phone number>"
client.publish(PhoneNumber=phone_number, Message='Test message'))
```

Expected:
No errors

Actual:
```
botocore.errorfactory.InvalidParameterValueException: An error occurred (ParameterValueInvalid) when calling the Publish operation: Could not find topic associated with phone number
```

Extra info:
The same code without Moto works and sends an SMS to my phone, also the AWS cli tool accepts only a phone number and message.

It seems moto tries to connect a phone number to a SNS topic, while this is not required by AWS. See https://github.com/spulec/moto/blob/master/moto/sns/responses.py#L304

**Comment by spulec** [comment]
Thanks; setting this as a feature request to allow phone numbers without topics.

**Comment by threewordphrase** [comment]
I am also running into this.

**Comment by jeffjen** [comment]
Running into this issue as well.  Is someone working on an issue?  If not what would be a good solution for this?

**Comment by jeffjen** [comment]
Found a workaround by creating a subscription with the number I need to send with in moto.

It seems like at this time, creating subscriptions do not require confirmation.

**Comment by bendog** [comment]
I'm also having this issue

**Comment by PabloLefort** [comment]
Same here

**Comment by kam193** [comment]
I just created a PR #3253 that should fix it. You can check if the behavior is consistent with SNS.

**Comment by bblommers** [comment]
This is now fixed as of moto >= 1.3.15.dev1027
